concordium-consensus/src/Concordium/ImportExport.hs

```diff
@@ -414,39 +415,41 @@ importBlocksV3 inFile firstGenIndex cbk = runExceptT $
     -- We handle all IO errors as serialization failures from a result perspective.
     onIOErr :: IOError -> ExceptT (ImportFailure a) m ()
     onIOErr = failWith . show
-    importSections hdl = do
+    importSections hdl fileSize = do
         eof <- liftIO $ hIsEOF hdl
         unless eof $ do
             sectionStart <- liftIO $ hTell hdl
-            sectionBS <- getLengthByteString hdl
+            sectionBS <- getLengthByteString hdl fileSize
             case decode sectionBS of
                 Left err -> failWith err
                 Right SectionHeader{..} -> do
                     when (sectionGenesisIndex >= firstGenIndex) $ do
                         replicateM_
                             (fromIntegral sectionBlockCount)
-                            ( importData hdl $
+                            ( importData hdl fileSize $
                                 ImportBlock sectionProtocolVersion sectionGenesisIndex
                             )
                         replicateM_
                             (fromIntegral sectionFinalizationCount)
-                            ( importData hdl $
+                            ( importData hdl fileSize $
                                 ImportFinalizationRecord sectionProtocolVersion sectionGenesisIndex
                             )
                     -- Move to the next section
                     liftIO $ hSeek hdl AbsoluteSeek (sectionStart + toInteger sectionLength)
-                    importSections hdl
-    importData :: Handle -> (BS.ByteString -> ImportData) -> ExceptT (ImportFailure a) m ()
-    importData hdl makeImport = do
-        blockBS <- getLengthByteString hdl
+                    importSections hdl fileSize
+    importData :: Handle -> Integer -> (BS.ByteString -> ImportData) -> ExceptT (ImportFailure a) m ()
```

Comment:
Can you please add a comment to this function documenting the meaning of the value and the reason for its existence.